### PR TITLE
charts: support configuring nodePort for the dashboard web

### PR DESCRIPTION
**Pull request by iawia002**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Support configuring nodePort for the dashboard web.
```

/cc @warjiang

### CODE REVIEW

**Comment by warjiang** [comment]
👍   When I reviewed the related PR this morning, I also thought about `nodePort` feature in chart.

File: charts/karmada-dashboard/templates/karmada-dashboard-web-service.yaml [hunk 1]
```diff
@@ -12,5 +12,8 @@ spec:
       targetPort: http
       protocol: TCP
       name: http
+      {{- with .Values.web.service.nodePort }}
```
Comment (warjiang) [inline charts/karmada-dashboard/templates/karmada-dashboard-web-service.yaml:15]: here we need more robust check, if user input 
```yaml
service:
    type: ClusterIP
    port: 8000
    nodePort: 32000
```
the render output will be :
```yaml
spec:
  type: ClusterIP
  ports:
    - port: 8000
      targetPort: http
      protocol: TCP
      name: http
      nodePort: 32000
```

so what about 
```
{{- if and (eq .Values.web.service.type "NodePort") (not (empty .Values.web.service.nodePort)) }}
nodePort: {{ .Values.web.service.nodePort }}
{{- end }}
```

**Comment by warjiang** [comment]
@iawia002 BTW, I've test it locally, it's ok. The only one thing we need to do is resolving the comment ~

File: charts/karmada-dashboard/templates/karmada-dashboard-web-service.yaml (hunk 1 above)
Comment (iawia002) [reply charts/karmada-dashboard/templates/karmada-dashboard-web-service.yaml]: Updated

**Pull request merged**